benchto-it/src/test/java/io/trino/benchto/integrationtest/BenchtoTrinoIntegrationTest.java

```diff
@@ -0,0 +1,161 @@
+import java.util.concurrent.TimeUnit;
+
+import static com.google.common.util.concurrent.Uninterruptibles.sleepUninterruptibly;
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = DriverApp.class)
+@org.springframework.boot.test.IntegrationTest("executionSequenceId=BEN_SEQ_ID")
+public class BenchtoTrinoIntegrationTest
+{
+    static PostgreSQLContainer<?> postgres;
+    static GenericContainer<?> service;
+    static GenericContainer<?> trino;
+
+    @Autowired
+    private ExecutionDriver executionDriver;
+
+    @Autowired
+    private BenchmarkProperties benchmarkProperties;
+
+    @Autowired
+    protected ApplicationContext context;
+
+    @BeforeClass
+    public static void setup()
+    {
+        Network network = Network.newNetwork();
+        startBenchtoService(network);
+        startTrino(network);
+    }
+
+    @Test
+    public void testBenchmark()
+    {
+        setBenchmark("test_benchmark");
+        executionDriver.execute();
+        verifyBenchmark();
+    }
+
+    private static void startBenchtoService(Network network)
+    {
+        postgres = new PostgreSQLContainer<>("postgres:11")
+                .withNetwork(network)
+                .withNetworkAliases("postgres");
+        postgres.start();
+        String jdbcUrl = format("jdbc:postgresql://%s:%d/%s?user=%s&password=%s",
+                "postgres",
+                postgres.getExposedPorts().get(0),
+                postgres.getDatabaseName(),
+                postgres.getUsername(),
+                postgres.getPassword());
+        service = new GenericContainer<>("prestodev/benchto-service:latest")
+                .withNetwork(network)
+                .withNetworkAliases("benchto-service")
+                .withEnv(ImmutableMap.of("SPRING_DATASOURCE_URL", jdbcUrl))
+                .withExposedPorts(8080)
+                .waitingFor(new HttpWaitStrategy()
+                        .forPort(8080)
+                        .forStatusCode(200));
+        service.start();
+        createEnvironment("TEST_ENV");
+        createTag("TEST_TAG", "TEST_ENV");
+        System.setProperty("test.service.host", service.getHost());
+        System.setProperty("test.service.port", service.getMappedPort(8080).toString());
+    }
+
+    private static void startTrino(Network network)
+    {
+        trino = new GenericContainer<>("trinodb/trino:363")
+                .withNetwork(network)
+                .withNetworkAliases("trino")
+                .withExposedPorts(8080)
+                .waitingFor(new HttpWaitStrategy()
+                        .forPort(8080)
+                        .forPath("/v1/info")
+                        .forStatusCode(200)
+                        .forResponsePredicate(response -> response.contains("\"starting\":false")));
+        trino.start();
+        // We sometimes get "No nodes available to run query"
```

Comment:
I wonder why we don't have such issue for `io.trino.tests.product.launcher.env.common.Standard#createPrestoContainer`.

Maybe we should actually execute `SELECT 1` instead?